### PR TITLE
[None][feat] Resubmission of the routing refactor in trtllmgen

### PR DESCRIPTION
**Pull request by ChristinaZ**
## Release Notes

* **New Features**
  * Added SigmoidRenorm routing method for MoE operations
  * Expanded support for float32 data types in routing inputs (bias and logits)
  * Enhanced SM90+ hardware optimization with cluster-based execution support

* **Improvements**
  * Refactored MoE routing system for improved flexibility and maintainability
  * Better routing method configuration with policy-driven architecture

## Description
This PR fixes the issues introduced by https://github.com/NVIDIA/TensorRT-LLM/pull/12246.
Previously, I had skipped the C++ unit tests and there were failing cases.
In this PR, I’ve addressed those failures and fixed the related bugs.

## Test Coverage
```
cd cpp/build
make -j$(nproc) google-tests
./tests/unit_tests/kernels/routingKernelsTest 

pytest tests/unittest/_torch/modules/moe/test_moe_module.py::test_fp32_routing_bias -v
```

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tests/unittest/_torch/modules/moe/test_moe_module.py [hunk 1]
```diff
@@ -1152,6 +1161,139 @@ def test_configurable_moe_single_gpu(
...
+    "quant_algo",
+    [
+        QuantAlgo.NVFP4,
+        QuantAlgo.FP8_BLOCK_SCALES,
+        QuantAlgo.W4A8_MXFP4_MXFP8,
+        QuantAlgo.FP8,
+    ],
+)
+@pytest.mark.parametrize("dtype", [torch.bfloat16])
+def test_fp32_routing_bias(routing_method_cls, moe_model_config, quant_algo, dtype):
```
Comment (xxi-nv) [inline tests/unittest/_torch/modules/moe/test_moe_module.py:1214]: Please rename it to highlight this is a enhanced TRTLLMGen test case.

Comment (xxi-nv) [reply tests/unittest/_torch/modules/moe/test_moe_module.py:1214]: And do you want to add it in the CI yml?

Comment (ChristinaZ) [reply tests/unittest/_torch/modules/moe/test_moe_module.py:1214]: Done. I think I don't need to add it in the CI yml.

**Review by xxi-nv — APPROVED** [review]

**Review by litaotju — APPROVED** [review]

**Review by longlee0622 — APPROVED** [review]
